IMP-/Small_StepT.thy

```diff
@@ -0,0 +1,179 @@
+ (*
+    Authors: Bilel Ghorbel, Florian Kessler
+    Based on the Big step semantics definition  with time of IMP
+ *)
+section "Small step semantics of IMP- "
+
+subsection "Small step semantics definition"
+theory Small_StepT  imports Main Big_StepT Rel_pow begin
+
+inductive
+  small_step :: "com * state \<Rightarrow> com * state \<Rightarrow> bool" (infix "\<rightarrow>" 55)
+where
+Assign:  "(x ::= a, s) \<rightarrow> (SKIP, s(x := aval a s))" |
+
+Seq1:    "(SKIP;;c\<^sub>2,s) \<rightarrow> (c\<^sub>2,s)" |
+Seq2:    "(c\<^sub>1,s) \<rightarrow> (c\<^sub>1',s') \<Longrightarrow> (c\<^sub>1;;c\<^sub>2,s) \<rightarrow> (c\<^sub>1';;c\<^sub>2,s')" |
+
+IfTrue:  "s b \<noteq> 0 \<Longrightarrow> (IF b \<noteq>0 THEN c\<^sub>1 ELSE c\<^sub>2,s) \<rightarrow> (c\<^sub>1,s)" |
+IfFalse: "s b = 0 \<Longrightarrow> (IF b \<noteq>0  THEN c\<^sub>1 ELSE c\<^sub>2,s) \<rightarrow> (c\<^sub>2,s)" |
+
+WhileTrue:   "s b \<noteq> 0 \<Longrightarrow> (WHILE b\<noteq>0 DO c,s) \<rightarrow>
+            (c ;; WHILE b \<noteq>0 DO c,s)" |
+WhileFalse:   "s b = 0 \<Longrightarrow> (WHILE b\<noteq>0 DO c,s) \<rightarrow>
+            (SKIP,s)"
+
+subsection "Transitive Closure"
+abbreviation
+  small_step_pow :: "com * state \<Rightarrow> nat \<Rightarrow> com * state \<Rightarrow> bool" ("_ \<rightarrow>* _ \<down> _" 55)
```

Comment:
the `→* t` looks weird to me.
what about `notation small_step_pow ("_ →⇗_⇖ _" 55)`
that's want one wants to read it, but maybe it is inconvenient to write.
maybe switch the bigstep notation to `notation big_step_t ("_ ⇒⇗_⇖ _" 55)` to be consistent?

Reply:
I have to admit. One of the reasons we didn't choose this notation, is that I personally didn't know how to type it! How silly of me! But I totally agree it is the one making most sense.

Reply:
done